Patches und Anpassungen an Testdatensätze mit verändertem Testdatentemplate

fmeineke opened the pull request:
Neu ist Split Excel und die die Ergänzung Klinische Dokumentation sowie UCUM Ergänzung gemäß KDS

Commit:
Medikation(2) entfernt

Commit:
eclipse config ignorieren

fmeineke committed:
Typo Prozedurentext

fmeineke committed:
Typo Prozedurentext

fmeineke committed:
Kommentar

fmeineke committed:
Added apache.poi for excel handling

fmeineke committed:
Klinische Dokumentation implementiert Laborbefund UCUM ergänzt

fmeineke committed:
Zerlegt Exceltemplates nach csv für converter

fmeineke committed:
"Private" Testroutine für Splitter

fmeineke commented:
Nur mit diesen Anpassungen können die xlsx aus https://tmfev.sharepoint.com/:f:/s/tmf/mi-i/EvI-7k5TvGlAtZGqjaY0LOUBlGLhRZRngkrX6WoUEdQtDQ?e=kj5H3c bearbeitet werden

alexanderkiel commented on src/main/java/de/uni_leipzig/life/csv2fhir/Csv2Fhir.java:
```diff
@@ -43,7 +43,7 @@ public Csv2Fhir(File inputDir, File outputFile) {
             put("Laborbefund.csv", new LaborbefundConverterFactory());
             put("Diagnose.csv", new DiagnoseConverterFactory());
             put("Prozedur.csv", new ProzedurConverterFactory());
-            put("Medikation (2).csv", new MedikationConverterFactory());
```
Kannst Du die Zeile bitte ganz entfernen. Wir werden die Medikation (2) ja nicht wieder benötigen.

alexanderkiel commented on pom.xml:
```diff
@@ -1,43 +1,54 @@
-    <artifactId>csv2fhir</artifactId>
-    <packaging>jar</packaging>
-    <version>1.0-SNAPSHOT</version>
+	<groupId>de.uni-leipzig.life.csv2fhir</groupId>
```
Wir verwenden üblicherweise keine Tabs. Es wäre schön, wenn Du das bei Dir in Eclipse anpassen könntest.

alexanderkiel commented on src/main/java/de/uni_leipzig/life/csv2fhir/converter/KlinischeDokumentationConverter.java:
```diff
@@ -1,23 +1,83 @@
+    private Quantity parseObservationValue() throws Exception {
+        try {
+            return new Quantity().setValue(DecimalUtil.parseDecimal(record.get("Wert")))
+            		.setSystem("http://unitsofmeasure.org").setCode("UCUM").setUnit(record.get("Einheit"));
```
UCUM ist keine Einheit. Wenn wir hier keine UCUM konforme Einheit haben, würde ich system und code lieber weglassen.

alexanderkiel commented on src/main/java/de/uni_leipzig/life/csv2fhir/converter/LaborbefundConverter.java:
```diff
@@ -77,7 +80,8 @@ private DateTimeType parseObservationTimestamp() throws Exception {
         try {
-            return new Quantity().setValue(DecimalUtil.parseDecimal(record.get("Messwert"))).setUnit(record.get("Einheit"));
+            return new Quantity().setValue(DecimalUtil.parseDecimal(record.get("Messwert")))
+            		.setSystem("http://unitsofmeasure.org").setCode("UCUM").setUnit(record.get("Einheit"));
```
Hier das Gleiche. UCUM ist keine Einheit.

alexanderkiel commented on src/test/java/fmeineke/imise/de/SplitMainTest.java:
```diff
@@ -0,0 +1,46 @@
+import de.uni_leipzig.life.csv2fhir.Csv2Fhir;
+import junit.framework.TestCase;
+
+public class SplitMainTest extends TestCase {
```
Entfern den manuellen Test bitte. Florian hat schon Code für ein Commandozeileninterface fertig. Damit geht das manuelle Testen auch einfach.

fmeineke committed:
Zahlreiche Bereinigungen im Hinblick auf KDS Kompatibilität und bessere
Fehlerverfolgung
- null Excepition bei Date Parsing
- Uhrzeitstunde auch einstellig
- Tabs gegen Spaces
- Ucum generierung und Erkennung in Eingabe
- Umlaute ansi / utf (Präparate)
- Einzeilige Exceptions
- neue Ucum Knverter Klasse

fmeineke committed:
Neu generierte CSV Beispiele

fmeineke committed:
Tabs 2 Spaces

fmeineke committed:
System für PZN und ASK

alexanderkiel commented:
Hallo Frank, soweit sieht alles gut aus. Schau aber bitte noch einmal, dass Du alle Dateien im UTF-8 Encoding speicherst. Mir zeigt IntelliJ mehrere Warnungen an und ich sehe kaputte Zeichen. Danke.

fmeineke committed:
Umstellung auf UTF für alle CSV Dateien

fmeineke committed:
Umstellung auf UTF

alexanderkiel commented:
Die CSV Dateien unter resources sind noch nicht in UTF-8.

fmeineke committed:
UTF CSV Testdaten

fmeineke commented:
Sollte auch gefixed sein – ich erzeuge jetzt die Daten direkt im resource Ordner.
 
Persönlich würde auch gerne andere Dinge ändern:
 
-          logger statt System.out
-          ein detaillierterer Checkmechanismus und darauf bessere Fehlermeldungen
-          Converter als Klasse mit record
-          Dopplungen wegbekommen (Messwerte einlesen, etc)
 
Aber das geht ja über einen Patch hinaus, berührt fast alle Klassen und wäre mit Florian / dir zu besprechen?
 
Anbei die Dokumente die ich Herrn Schepers geschickt habe.
 
Bekommst Du denn immer Nachricht, wenn ich den Push HEAD klicke?
 
Gruß, Frank 
 
Von: Alexander Kiel <notifications@github.com> 
Gesendet: Donnerstag, 5. November 2020 13:00
An: life-research/csv2fhir <csv2fhir@noreply.github.com>
Cc: fmeineke <frank.meineke@imise.uni-leipzig.de>; Author <author@noreply.github.com>
Betreff: Re: [life-research/csv2fhir] Patches und Anpassungen an Testdatensätze mit verändertem Testdatentemplate (#2)
 
Die CSV Dateien unter resources sind noch nicht in UTF-8.
—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub https://github.com/life-research/csv2fhir/pull/2#issuecomment-722333633, or unsubscribe https://github.com/notifications/unsubscribe-auth/AENL7TKZUGSON7PAE4LAWJTSOKHSLANCNFSM4THYK3VA.
